compiler/typecheck/TcErrors.hs

```diff
@@ -2404,6 +2405,28 @@ mkDictErr ctxt cts
     -- but we really only want to report the latter
     elim_superclasses cts = mkMinimalBySCs ctPred cts
 
+customDamlErrors :: Ct -> [ClsInst] -> SDoc -> Maybe SDoc
```

Comment:
yes this is great!

Reply:
After PoC, I think it would be nice to have this in its own file `compiler/typecheck/TcErrors/Daml.hs`

Reply:
Going forward I think this module might end up being used in more places than just `TcErrors`, so I've opted to put it at the top level as `compiler/typecheck/TcDaml.hs` instead